tests/bsim/bluetooth/host/adv/resume2/dut/main.c

```diff
@@ -0,0 +1,209 @@
+
+	/* `bt_le_adv_start` is invoked once, and.. */
+
+	err = bt_le_adv_start(BT_LE_ADV_CONN_NAME_AD, NULL, 0, NULL, 0);
+	__ASSERT_NO_MSG(!err);
+
+	/* .. the advertiser shall autoresume. Since it's not
+	 * stopped, it shall continue receivng connections until
+	 * the stack runs out of connection objects.
+	 */
+
+	LOG_INF("Waiting for connections...");
+	while (atomic_get(&connected_count) < CONFIG_BT_MAX_CONN) {
+		k_msleep(1000);
+	}
+
+	LOG_INF("✅ Ok");
+
+
+	LOG_INF("👉 Disconnect one to see that it comes back");
+
+	/* Disconnect one of the connections. It does matter
+	 * which, but object with index 0 is chosen for
+	 * simplicity.
+	 */
+
+	conn = bt_testlib_conn_unindex(BT_CONN_TYPE_LE, 0);
+	__ASSERT_NO_MSG(conn);
+
+	/* Disconnect, but wait with unreffing.. */
+
+	err = bt_conn_disconnect(conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+	__ASSERT_NO_MSG(!err);
+	bt_testlib_wait_disconnected(conn);
+
+	/* Simulate a delayed unref. We delay to make sure the resume is not
+	 * triggered by disconnection, but by a connection object becoming
+	 * available.
+	 */
+
+	k_sleep(K_SECONDS(10));
+
+	bt_testlib_conn_unref(&conn);
+
+	/* Since there is a free connection object again, the
+	 * advertiser shall automatically resume and receive a
+	 * new connection.
+	 */
+
+	LOG_INF("Waiting for connections...");
+	while (atomic_get(&connected_count) < CONFIG_BT_MAX_CONN) {
+		k_msleep(1000);
+	}
+
+	LOG_INF("✅ Ok");
+
+
+	LOG_INF("🧹 Clean up");
+
+	err = bt_le_adv_stop();
+	__ASSERT_NO_MSG(!err);
+
+	disconnect_all();
+
+	LOG_INF("✨ Ok");
+
+
+	LOG_INF("🚧 Setup: Connect one central connection");
+
+	err = bt_testlib_scan_find_name(&connectable_addr, "connectable");
+	__ASSERT_NO_MSG(!err);
+
+	err = bt_testlib_connect(&connectable_addr, &conn);
+	__ASSERT_NO_MSG(!err);
+
+	LOG_INF("✅ Ok");
+
+
+	LOG_INF("🚧 Setup: Start advertiser. Let it fill the connection limit.");
+
+	/* With one connection slot taken by the central role, we fill the rest. */
+
+	err = bt_le_adv_start(BT_LE_ADV_CONN_NAME_AD, NULL, 0, NULL, 0);
+	__ASSERT_NO_MSG(!err);
+
+	LOG_INF("Waiting for connections...");
+	while (atomic_get(&connected_count) < CONFIG_BT_MAX_CONN) {
+		k_sleep(K_SECONDS(1));
+	}
+
+	LOG_INF("✅ Ok");
+
+
+	LOG_INF("👉 Main test: Disconnect, wait and connect the central connection.");
+
+	/* In this situation, disconnecting the central role should not allow
+	 * the advertiser to resume. This behavior was introduced in 372c8f2d92.
+	 */
+
+	LOG_INF("🚩 Disconnect");
+	err = bt_testlib_disconnect(&conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+	__ASSERT_NO_MSG(!err);
+
+	LOG_INF("🚩 Wait to bait the advertiser");
+	k_sleep(K_SECONDS(5));
+
+	LOG_INF("💣 Connect");
+	err = bt_testlib_connect(&connectable_addr, &conn);
+	if (err) {
+		/* If the test fails, it's because the advertiser 'stole' the
+		 * central's connection slot.
+		 */
+		__ASSERT_NO_MSG(err == -ENOMEM);
+
+		LOG_ERR("💥 Advertiser stole the connection slot");
```

Comment:
internal: we should do a last test run to see if it still fails with `372c8f2d92` reverted.

Reply:
Verified. Fails with `💥 Advertiser stole the connection slot`.